Makefile

```diff
@@ -75,4 +80,45 @@ deploy-gatekeeper:
 
 .PHONY: delete-gatekeeper
 delete-gatekeeper:
-	helm delete gatekeeper --namespace gatekeeper-system 
\ No newline at end of file
+	helm delete gatekeeper --namespace gatekeeper-system 
+
+.PHONY: test-e2e
+test-e2e:
+	bats -t ${BATS_TESTS_FILE}
+
+e2e-dependencies:
+	# Download and install kind
+	curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output ${GITHUB_WORKSPACE}/bin/kind && chmod +x ${GITHUB_WORKSPACE}/bin/kind
+	# Download and install kubectl
+	curl -L https://storage.googleapis.com/kubernetes-release/release/v${KUBERNETES_VERSION}/bin/linux/amd64/kubectl -o ${GITHUB_WORKSPACE}/bin/kubectl && chmod +x ${GITHUB_WORKSPACE}/bin/kubectl
+	# Download and install bats
+	curl -sSLO https://github.com/bats-core/bats-core/archive/v${BATS_VERSION}.tar.gz && tar -zxvf v${BATS_VERSION}.tar.gz && bash bats-core-${BATS_VERSION}/install.sh ${GITHUB_WORKSPACE}
+
+KIND_NODE_VERSION := kindest/node:v$(KUBERNETES_VERSION)
+
+e2e-bootstrap: e2e-dependencies
+	# Check for existing kind cluster
+	if [ $$(${GITHUB_WORKSPACE}/bin/kind get clusters) ]; then ${GITHUB_WORKSPACE}/bin/kind delete cluster; fi
+	# Create a new kind cluster
+	TERM=dumb ${GITHUB_WORKSPACE}/bin/kind create cluster --image $(KIND_NODE_VERSION) --wait 5m
+
+e2e-helm-install:
+	rm -rf .staging/helm
+	mkdir -p .staging/helm
+	curl https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz > .staging/helm/helmbin.tar.gz
```

Comment:
nit: consider standardizing how curl output is saved here.  It is seems to use --output, -o and > to achieve the same thing

Reply:
thanks , fixed!